Sources/AMQPClient/AMQPClient.swift

```diff
@@ -68,20 +81,27 @@ public final class AMQPClient {
     /// - Returns: EventLoopFuture with result confirming that broker has accepted a request.
     @discardableResult
     public func connect() ->  EventLoopFuture<AMQPResponse.Connection.Connected> {
-        guard self.connection == nil else { return self.eventLoopGroup.any().makeFailedFuture(AMQPClientError.alreadyConnected) }
+        self.lock.lock()
+        defer { self.lock.unlock() }
 
-        guard self.isConnect.compareExchange(expected: false, desired: true, ordering: .relaxed).exchanged else {
-            return self.eventLoopGroup.any().makeFailedFuture(AMQPClientError.alreadyConnecting)
+        guard case .disconnected = self._connState else {
+            return self.eventLoopGroup.any().makeFailedFuture(AMQPClientError.alreadyConnect)
         }
 
+        self._connState = .connecting
```

Comment:
This holds the lock for to long. Use `self.lock.withLock {}` instead!

Reply:
ok check now :)